quinn-proto/src/connection/streams/mod.rs

```diff
@@ -254,10 +254,10 @@ impl<'a> SendStream<'a> {
     /// # Panics
     /// - when applied to a receive stream
     pub fn reset(&mut self, error_code: VarInt) -> Result<(), UnknownStream> {
-        let stream = match self.state.send.get_mut(&self.id) {
-            Some(ss) => ss,
-            None => return Err(UnknownStream { _private: () }),
-        };
+        let stream = self
```

Comment:
It's not that important, but I don't think I'd count this as an improvement; basically I think my argument would be is that this is the same number of lines but with all the density on a single line. By comparison, the `match` version spreads the density around between "what are we matching" and "what do we do if we can't find the thing we're looking for", and makes it more visually obvious that there are two alternatives.

Reply:
I appreciate the readability of a standard combinator, personally; it's very easy to interpret at a glance, at least if you use a lot of `ok_or`, which I find that I do.